packages/drivers/driver-base/src/driverUtils.ts

```diff
@@ -128,16 +129,38 @@ export function validateMessages(
 			messages.length = 0;
 		}
 		if (last + 1 !== from + length) {
+			// If not strict, then return the first consecutive sub-block.
+			if (strict) {
+				messages.length = 0;
+			} else {
+				let validOpsCount = 1;
+				while (
+					validOpsCount < messages.length &&
+					messages[validOpsCount].sequenceNumber ===
+						messages[validOpsCount - 1].sequenceNumber + 1
+				) {
+					validOpsCount += 1;
+				}
+				messages.length = Math.min(messages.length, validOpsCount);
```

Comment:
You do not need to do min() here, validOpsCount is guaranteed to be not higher.

Reply:
In corner case when the messages only had 0/1 ops and validOpsCount start with 1. So I did the min here.

Reply:
If length == 0, then we would not get here (check on line 115). In all other cases validOpsCount would be less or equal to messages.length (condition on like 127). So I think you can safely remove min()

Reply:
Created PR here: https://github.com/microsoft/FluidFramework/pull/16206
It was required previously when we had separate ifs :)